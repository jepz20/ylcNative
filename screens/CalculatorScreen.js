// @flow
import React, { Component } from 'react'
import { View, StyleSheet, LayoutAnimation } from 'react-native'
import { connect } from 'react-redux'
import { DigitsContainer, ControllersContainer } from '../containers'
import * as actions from '../actions'
import type { StateRE } from '../reducers/calculator'
import type { DigitPress, Erase, Reset } from '../actions/calculator'
type Props = {
  calculator: StateRE,
  digitPress: DigitPress,
  erase: Erase,
  reset: Reset
}
export class CalculatorScreen extends Component<Props> {
  static defaultProps = {
    calculator: {}
  }

  componentDidMount () {
    console.log('I was mounted', this.props.calculator)
  }

  componentWillUpdate (nextProps: Props) {
    console.log('here', nextProps, this.props, nextProps.calculator)

    if (nextProps.calculator.visible === this.props.calculator.visible) {
      return
    }
    LayoutAnimation.easeInEaseOut()
  }

  render () {
    if (!this.props.calculator.visible) return null

    const { digitPress, erase, calculator, reset } = this.props

    return (
      <View style={styles.container}>
        <ControllersContainer player={calculator.currentPlayer} />
        <DigitsContainer
          value={calculator.value}
          digitPress={digitPress}
          erase={erase}
          reset={reset}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    position: 'absolute',
    right: 0,
    left: 0,
    top: 0,
    bottom: 0,
    marginTop: 60
  }
})

const mapStateToProps = ({ calculator }) => ({
  calculator
})

export default connect(mapStateToProps, actions)(CalculatorScreen)
