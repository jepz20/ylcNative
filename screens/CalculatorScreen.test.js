// __tests__/Intro-test.js
import 'react-native'
import React from 'react'
import { CalculatorScreen } from './CalculatorScreen'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

test('renders empty is calculator is not visible', () => {
  const tree = renderer.create(<CalculatorScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders if the calculator is visible', () => {
  const tree = renderer
    .create(<CalculatorScreen calculator={{ visible: true }} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
