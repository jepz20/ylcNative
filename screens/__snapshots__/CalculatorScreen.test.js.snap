// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty is calculator is not visible 1`] = `null`;
